src/sqlfluff/dialects/dialect_duckdb.py

```diff
@@ -23,6 +23,14 @@
 
 duckdb_dialect = postgres_dialect.copy_as("duckdb")
 
+duckdb_dialect.replace(
+    SingleIdentifierGrammar=OneOf(
+        Ref("NakedIdentifierSegment"),
+        Ref("QuotedIdentifierSegment"),
+        Ref("QuotedLiteralSegment"),
```

Comment:
I think this should be this, to keep it as an identifier as it would be with double quoted identifier (and so allow appropriate rules to match):

```suggestion
        Ref("SingleQuotedIdentifierSegment"),
```